### PR TITLE
IA-2881 IA-2878 Update to latest wb-libs

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-2881
https://broadworkbench.atlassian.net/browse/IA-2878

Depends on https://github.com/broadinstitute/workbench-libs/pull/698

* Made `LeoPubsubMessageSubscriber` depends on `GKEAlgebra[F]` instead of `GKEInterpreter[F]` (this will prevent we rely on implementation details of `GKEInterpreter[F]` in `LeoPubsubMessageSubscriberSpec`). Moved some unit tests in `LeoPubsubMessageSubscriberSpec` that essentially testing `GKEInterpreter`'s behavior to `GKEInterpreterSpec`. This should make these tests a lot more focused on what they're trying to validate.

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by Qi77Qi** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
leonardo build failed..jenkins retest

**Comment by Qi77Qi** [comment]
leonardo build failed..jenkins retest

**Comment by Qi77Qi** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
fiab-start failure...jenkins retest

**Comment by Qi77Qi** [comment]
multi-test -- one of the run faild due to fiab-start failure

**Comment by Qi77Qi** [comment]
multi-test -- one of the run failed with a selenium test

**Comment by Qi77Qi** [comment]
multi-test...2 failed, one failed due to jenkins node ran out of space, another failed with `listApp` getting 401

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
jenkins retest

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSuite.scala [hunk 1]
```diff
@@ -167,63 +169,65 @@ trait GPAllocBeforeAndAfterAll extends GPAllocUtils with BeforeAndAfterAll {
...
-            loggerIO.info(s"Created initial runtime ${project.value} / ${initalRuntimeName.asString}")
-          case Left(err) =>
-            loggerIO
-              .warn(err)(
-                s"Failed to create initial runtime ${project.value} / ${initalRuntimeName.asString} with error"
-              )
-        }
-      } yield ()
-    }
+    if (isHeadless) {
```
Comment (rtitle) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSuite.scala:185]: This means we only create the initial runtime for Jenkins runs, not local, right? That makes sense I think

Comment (Qi77Qi) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSuite.scala:185]: yep yep...I'm commenting it out every time I run it locally, so figured I'll make it a bit simpler :)..sry missed this in description

File: http/src/main/resources/logback.xml [hunk 2]
```diff
@@ -41,7 +36,7 @@
         <appender-ref ref="SYSLOG"/>
     </logger>
 
-    <logger name="akka" level="info" additivity="false">
+    <logger name="akka" level="debug" additivity="false">
```
Comment (rtitle) [inline http/src/main/resources/logback.xml]: Change back to `info`? (Same for http4s below)

Comment (Qi77Qi) [reply http/src/main/resources/logback.xml]: ah yes!

File: http/src/main/resources/reference.conf [hunk 3]
```diff
@@ -504,14 +504,14 @@ clusterFiles {
 # so we need to ensure we don't use stale cache entries.
 runtimeDnsCache {
   cacheExpiryTime = 5 seconds
-  cacheMaxSize = 10000
+  cacheMaxSize = 1000
```
Comment (rtitle) [inline http/src/main/resources/reference.conf:507]: Is there a reason to reduce these? I figure these caches don't use a ton of memory, and it's good to be able to support a lot of users without evictions.

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SamAuthProvider.scala [hunk 4]
```diff
@@ -1,75 +1,46 @@
...
-      blocker.blockOn(
-        Effect[F].delay(
-          authCache
-            .get(
-              AuthCacheKey(sr.resourceType, sr.resourceIdAsString(samResource), authHeader, sr.actionAsString(action))
-            )
-            .booleanValue()
-        )
-      )
+      authCache.cachingF(
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SamAuthProvider.scala:39]: The `scalacache` syntax is nicer than guava for sure

File: project/Dependencies.scala [hunk 5]
```diff
@@ -10,28 +10,27 @@ object Dependencies {
...
-  val http4sVersion = "0.21.26"
+  val http4sVersion = "0.23.5"
   val guavaV = "31.0.1-jre"
   val monocleV = "2.1.0"
   val opencensusV = "0.28.3"
 
 
-  private val workbenchLibsHash = "bf66e61"
-  val serviceTestV = s"0.19-$workbenchLibsHash"
+  private val workbenchLibsHash = "6bb15228-SNAP"
```
Comment (rtitle) [inline project/Dependencies.scala]: Update to non SNAP before merge

**Review by rtitle — APPROVED** [review]
A lot of changes but honestly it all looks good if tests (and multi tests) pass. Interesting learning about the new cats-effect API. Thanks for doing this!

File: http/src/main/resources/reference.conf (hunk 3 above)
Comment (Qi77Qi) [reply http/src/main/resources/reference.conf:507]: Screen Shot 2021-10-10 at 11 56 35 AM

I looked at prod metrics over last 6 weeks, and for `runtimeDnsCache`, it seems maxed around 15 for size. So I think 1000 should be enough.

`googleTokenCache` has the largest size (see [dashboard](https://console.cloud.google.com/monitoring/dashboards/builder/813abbd0-604d-473c-a958-dd2dc205490d?authuser=0&project=broad-dsde-prod&dashboardBuilderState=%257B%2522editModeEnabled%2522:false%257D&timeDomain=6w) )...I'll make that larger then 1k

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
@rtitle thanks for reviews....think I've address your comments..lmk if you have more...

**Pull request merged**